code/modules/admin/verbs/playsound.dm

```diff
@@ -2,55 +2,74 @@
 	set category = "Fun"
 	set name = "Play Imported Sound"
 	set desc = "Play a sound imported from anywhere on your computer."
-	if(!check_rights(R_SOUNDS))	return
+	if(!check_rights(R_SOUNDS))
+		return
 
-	if(midi_playing)
-		to_chat(usr, "No. An Admin already played a midi recently.")
+	var/vol = input(usr, "What volume would you like the sound to play at?",, 100) as null|num
+	if(!vol)
+		return
+	if(vol < 0 || vol > 100) //clamp when?
+		to_chat(src, "<span class='notice'>Invalid volume!</span>")
 		return
 
-	var/sound/uploaded_sound = sound(S, repeat = 0, wait = 1, channel = 777)
-	uploaded_sound.priority = 250
-
-	switch( alert("Play sound globally or locally?", "Sound", "Global", "Local", "Cancel") )
-		if("Global")
-			for(var/mob/M in player_list)
-				if(M.client.prefs.toggles_sound & SOUND_MIDI)
-					M << uploaded_sound
-					heard_midi++
-		if("Local")
-			playsound(get_turf(src.mob), uploaded_sound, 50, 0)
-			for(var/mob/M in view())
-				heard_midi++
+	var/sound/admin_sound = new()
+	admin_sound.priority = 250
+	admin_sound.file = S
+	admin_sound.volume = vol
+	admin_sound.channel = 777
+
+	var/res = alert(usr, "Show the title of this song to the players?",, "Yes","No", "Cancel")
```

Comment:
Just because I'm curious and it's relevant to something else I'm working on, the empty parameter will make it use the default? I saw the compiler didn't throw any warnings.